internal/tracer/otel.go

```diff
@@ -0,0 +1,33 @@
+package tracer
+
+import (
+	"context"
+
+	"github.com/windmilleng/wmclient/pkg/dirs"
+	"go.opentelemetry.io/otel/api/global"
+	apitrace "go.opentelemetry.io/otel/api/trace"
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+)
+
+var tracer apitrace.Tracer
+
+func InitOpenTelemetry(ctx context.Context, dir *dirs.WindmillDir) (Locker, error) {
+	exporter, err := newExporter(ctx, dir)
+	if err != nil {
+		return nil, err
+	}
+
+	tp, err := sdktrace.NewProvider(
+		sdktrace.WithConfig(sdktrace.Config{DefaultSampler: sdktrace.AlwaysSample()}))
+	if err != nil {
+		return nil, err
+	}
+	tp.RegisterSpanProcessor(exporter)
+	global.SetTraceProvider(tp)
+	tracer = tp.Tracer("tilt.dev/usage") // global
```

Comment:
why is this global?

afaict, it should be straightforward to dependency-inject the tracer. global state is going to fall apart pretty quickly

Reply:
Just for similarity to opentracing, where we use a global tracer and haven't had an issue.

Reply:
?

in opentracing, you usually inject the tracer with server middleware, see this example:
https://github.com/grpc-ecosystem/go-grpc-middleware/blob/master/tracing/opentracing/client_interceptors.go#L21
https://github.com/grpc-ecosystem/go-grpc-middleware/blob/master/tracing/opentracing/options.go#L51

it looks like opentracing is encouraging people to use global.TraceProvider().Trace() when you need a new tracer, see this example:
https://github.com/open-telemetry/opentelemetry-go/blob/master/example/grpc/middleware/tracing/tracing.go#L51

```
tr := global.TraceProvider().Tracer("example/grpc")
```

can you do that instead?